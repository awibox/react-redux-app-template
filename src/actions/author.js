import axios from 'axios';
import {GET_AUTHOR_SUCCESS} from '../actions/types';
import {getError} from "./error";
import {gitHubApiUrl, authorAccount} from '../../configs/config';

const END_POINT = 'users';

const getAuthorSuccess = author => ({
    type: GET_AUTHOR_SUCCESS,
    payload: author
});

export const getAuthor = () => dispatch => {
    axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}`)
        .then(response => {
            const {data} = response;
            dispatch(getAuthorSuccess(data));
        }).catch(err => {
            dispatch(getError(err.response.data))
        });
};
