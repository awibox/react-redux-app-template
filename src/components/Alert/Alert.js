import React from 'react';
import PropTypes from 'prop-types';
// Styles
import styles from './Alert.scss';

const Alert = (props) => (
  <div className={styles.alert}>
    {props.children}
  </div>
);

Alert.propTypes = {
  children: PropTypes.any,
};

export default Alert;
