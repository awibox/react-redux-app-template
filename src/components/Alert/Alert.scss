@import '../../styles/variables';

.alert {
  padding: $defaultIndent;
  background: #f47f7f;
  color: #fff;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(.25, .8, .25, 1);
  border-radius: 3px;
  margin-bottom: $defaultIndent;
}
