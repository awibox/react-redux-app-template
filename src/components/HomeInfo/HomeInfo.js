import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
// Components
import { Button, Card } from 'tigerspack';
// Styles
import styles from './HomeInfo.scss';

// Using "Stateless Functional Components"
export default class HomeInfo extends Component {
  static propTypes = {
    home: ImmutablePropTypes.contains({
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired,
    }),
  };

  buttonClick = (url) => window.open(url);

  render() {
    const { home } = this.props;
    return (
      <div className={styles.homeInfo}>
        <Card theme={'light'}>
          <div className={styles.homeLogo}>
            <div className={styles.image}>
              <i className="fa fa-gears"></i>
            </div>
            <h1 className={styles.name}>{home.get('name')}</h1>
            <div className={styles.text}>{home.get('description')}</div>
            <div className={styles.button}>
              <Button icon={<i className="fa fa-github"/>} onClick={() => this.buttonClick(home.get('html_url'))}>
                Download with Github
              </Button>
            </div>
          </div>
        </Card>
      </div>
    );
  }
}
