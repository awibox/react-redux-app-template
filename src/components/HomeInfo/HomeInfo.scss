@import '../../styles/variables';


.homeInfo {
  text-align: center;
}

.homeLogo {
  display: flex;
  flex-flow: column;
  .image {
    font-size: 128px;
    padding: $defaultIndent;
    color: $logoColor;
  }
  .name {
    line-height: 56px;
    font-size: 56px;
    font-weight: 100;
    padding-top: 10px;
  }
  .text {
    padding: 20px;
    font-size: 20px;
    color: #333;
    font-weight: 100;
    line-height: 30px;
    margin-bottom: 30px;
  }
}

.version {
  font-size: 18px;
  line-height: 22px;
  text-align: left;
  margin-left: 5px;
}

.button {
  padding-bottom: 20px;
}
