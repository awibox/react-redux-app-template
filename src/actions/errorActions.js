import { GET_ERRORS } from 'actions/types';

export const getError = (error) => ({
  type: GET_ERRORS,
  payload: error,
});

export default getError;
