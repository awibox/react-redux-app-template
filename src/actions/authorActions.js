import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';
import { GET_AUTHOR_SUCCESS } from 'actions/types';
import { gitHubApiUrl, authorAccount } from 'config';
import { getError } from 'actions/errorActions';


const END_POINT = 'users';

export const getAuthorSuccess = (author) => ({
  type: GET_AUTHOR_SUCCESS,
  payload: author,
});

export const getAuthorAction = () => async (dispatch) => {
  try {
    const { data } = await trackPromise(axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}`));
    dispatch(getAuthorSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
