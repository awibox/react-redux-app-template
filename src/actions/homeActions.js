import axios from 'axios';
import { trackPromise } from 'react-promise-tracker';
import { GET_HOME_SUCCESS } from 'actions/types';
import { gitHubApiUrl, authorAccount, repoName } from 'config';
import { getError } from './errorActions';

const END_POINT = 'repos';

export const getHomeSuccess = (home) => ({
  type: GET_HOME_SUCCESS,
  payload: home,
});

export const getHomeAction = () => async (dispatch) => {
  try {
    const { data } = await trackPromise(axios.get(`${gitHubApiUrl}/${END_POINT}/${authorAccount}/${repoName}`));
    dispatch(getHomeSuccess(data));
  } catch (err) {
    dispatch(getError(err.response.data));
  }
};
