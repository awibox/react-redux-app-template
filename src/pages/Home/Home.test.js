import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import HomeInfo from 'components/HomeInfo/HomeInfo';
import Alert from 'components/Alert/Alert';
import { Map } from 'immutable';
import { getHomeSuccess } from 'actions/homeActions';
import HomeContainer from './Home';

describe('Home', () => {
  const mockData = Map({
    name: 'Finder',
    description: 'Search engine',
    html_url: 'http://localhost:8888/url',
  });
  const initialState = {
    homeState: {
      home: mockData,
    },
    errors: {
      message: 'Fake error',
    },
    load: {
      isLoading: false,
    },
    getHomeAction: () => {},
  };
  const mockStore = configureStore([thunk]);
  let store; let
    container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = mount(<Provider store={store}>
      <HomeContainer getHomeAction={initialState.getHomeAction}/>
    </Provider>);
  });
  it('should render corectly', () => {
    expect(container).toMatchSnapshot();
  });
  it('should render the container', () => {
    expect(container.length).toEqual(1);
  });
  it('should matches with initialState in the HomeInfo', () => {
    expect(container.find(HomeInfo).prop('home')).toEqual(mockData);
  });
  it('should display Alert', () => {
    expect(container.find(Alert)).toExist();
  });
  it('should check prop Alert', () => {
    expect(container.find(Alert).prop('children')).toEqual(initialState.errors.message);
  });
  it('should action on dispatching', () => {
    store.dispatch(getHomeSuccess(mockData));
    const action = store.getActions();
    expect(action[0].type).toBe('GET_HOME_SUCCESS');
    expect(action[0].payload).toEqual(mockData);
    container.unmount();
  });
});
