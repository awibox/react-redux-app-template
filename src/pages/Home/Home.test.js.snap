// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render corectly 1`] = `
<Connect(HomeContainer)
  getHomeAction={[Function]}
>
  <div>
    <div
      className="alert"
    >
      Fake error
    </div>
    <div
      className="homeInfo"
    >
      <div
        className="card"
      >
        <div
          className="homeLogo"
        >
          <div
            className="image"
          >
            <i
              className="fa fa-gears"
            />
          </div>
          <h1
            className="name"
          >
            Finder
          </h1>
          <div
            className="text"
          >
            Search engine
          </div>
          <div
            className="button"
          >
            <div
              className="btn btnIcon"
              onClick={[Function]}
            >
              <span
                className="icon"
              >
                <i
                  className="fa fa-github"
                />
              </span>
              Download with Github
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="loader"
    >
      <div
        className="spinnerBorder"
      />
    </div>
  </div>
</Connect(HomeContainer)>
`;
